salon_creator/lib/screens/user_profile_edit_screen.dart

```diff
@@ -0,0 +1,320 @@
+import 'dart:io';
+
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:firebase_storage/firebase_storage.dart';
+import 'package:flutter/material.dart';
+import 'package:image_picker/image_picker.dart';
+import 'package:salon_creator/common/color.dart';
+import 'package:salon_creator/constant/constants.dart' as constants;
+import 'package:salon_creator/models/user_model.dart';
+import 'package:uuid/uuid.dart';
+
+class UserProfileEditScreen extends StatefulWidget {
+  const UserProfileEditScreen({
+    Key key,
+    this.userModel,
+  }) : super(key: key);
+
+  final UserModel userModel;
+
+  @override
+  _UserProfileEditScreenState createState() => _UserProfileEditScreenState();
+}
+
+class _UserProfileEditScreenState extends State<UserProfileEditScreen> {
+  final ImagePicker imagePicker = ImagePicker();
+  TextEditingController nameController;
+  TextEditingController aboutMeController;
+  bool _opInProgress = false;
+
+  bool _profileUpdated = false;
+  XFile _imageFile;
+
+  void _updateScreenContext(UserModel userModel) {
+    var _hasProfileChanged = userModel.profile.name != nameController.text ||
+        userModel.profile.aboutMeText != aboutMeController.text ||
+        _imageFile != null;
+
+    setState(() {
+      _profileUpdated = _hasProfileChanged;
+    });
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    nameController = TextEditingController(text: widget.userModel.profile.name);
+    aboutMeController =
+        TextEditingController(text: widget.userModel.profile.aboutMeText);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return StreamBuilder<DocumentSnapshot<Map<String, dynamic>>>(
+      stream: FirebaseFirestore.instance
+          .collection(constants.DBCollection.users)
+          .doc(FirebaseAuth.instance.currentUser.email)
+          .snapshots(),
+      builder: (context, snapshot) {
+        if (!snapshot.hasData) {
+          return LinearProgressIndicator();
+        }
+
+        var userModel = UserModel.fromMap(snapshot.data.data());
+
+        return _buildUserProfileEditScreenInternal(userModel);
+      },
+    );
+  }
+
+  Future<bool> _onBackButtonPressed() async {
+    return !_profileUpdated
+        ? true
+        : await showDialog(
+              context: context,
+              builder: (context) => new AlertDialog(
+                title: new Text('変更内容を保存しますか？'),
+                content: new Text('変更内容を保存して、後で編集を続けられるようにしますか'),
```

Comment:
I would say "All changes will be lost" here.
```suggestion
                content: Text('変更内容は全て失われます'),
```